js/filteringBox.js

```diff
@@ -27,6 +27,52 @@ function openFilterModal() {
     closeFilterButton.classList.add("filter__box__topDiv__closeFilterBtn");
     closeFilterButton.innerHTML = "X";
     closeFilterButton.addEventListener("click", closeFilterModal);
+
+    // Checkbox divs
+    const checkDivMain = document.createElement("div");
+    filterBox.appendChild(checkDivMain);
+    checkDivMain.classList.add("filter__box__checkDivMain");
+
+    const checkDiv2 = document.createElement("div");
+    checkDivMain.appendChild(checkDiv2);
+    checkDiv2.classList.add("filter__box__checkDivMain__checkDiv2");
+
+    const checkDiv3 = document.createElement("div");
+    checkDivMain.appendChild(checkDiv3);
+    checkDiv3.classList.add("filter__box__checkDivMain__checkDiv3");
+
+    const checkDiv4 = document.createElement("div");
+    checkDivMain.appendChild(checkDiv4);
+    checkDiv4.classList.add("filter__box__checkDivMain__checkDiv4");
+
+    // Checkboxes
+    const checkBoxTitle = document.createElement("h4");
+    checkBoxTitle.innerHTML = "By type";
+    checkDiv2.appendChild(checkBoxTitle);
+    checkBoxTitle.classList.add("filter__box__checkDiv__title");
+
+    const checkBox1 = document.createElement("input");
+    checkBox1.type = ("checkbox");
+    checkBox1.id = "DOM__checkBox1";
+    checkDiv3.appendChild(checkBox1);
+    checkBox1.classList.add("filter__box__checkDiv__checkbox1");
+
+    const checkBoxLabel1 = document.createElement("label");
+    checkBoxLabel1.htmlFor = "DOM__checkBox1";
+    checkBoxLabel1.appendChild(document.createTextNode("Include online challenges"));
+    checkDiv3.appendChild(checkBoxLabel1);
+     
+    const checkBox2 = document.createElement("input");
+    checkBox2.type = ("checkbox");
+    checkBox2.id = ("DOM__checkBox2")
+    checkDiv4.appendChild(checkBox2);
+    checkBox2.classList.add("filter__box__checkDiv__checkbox2");
+
+    const checkBoxLabel2 = document.createElement("label");
+    checkBoxLabel2.htmlFor = "DOM__checkBox1";
```

Comment:
The htmlFor attribute of the second label is set to "DOM__checkBox1" instead of "DOM__checkBox2". This results in an incorrect association with the checkbox. :)